pkg/apis/batch/validation/validation.go

```diff
@@ -259,6 +264,13 @@ func validateJobSpec(spec *batch.JobSpec, fldPath *field.Path, opts apivalidatio
 	if spec.PodFailurePolicy != nil {
 		allErrs = append(allErrs, validatePodFailurePolicy(spec, fldPath.Child("podFailurePolicy"))...)
 	}
+	if spec.SuccessPolicy != nil {
+		if ptr.Deref(spec.CompletionMode, batch.NonIndexedCompletion) == batch.NonIndexedCompletion {
```

Comment:
```suggestion
		if ptr.Deref(spec.CompletionMode, batch.NonIndexedCompletion) != batch.IndexedCompletion {
```

If we add a new completionMode, this will keep validation as we currently intend it, right?

Reply:
It makes sense.

Reply:
Done.